4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AnimalShelterCode
+{
+    public class AnimalShelter
+    {
+        public int AnimalCounter { get; private set; }
+
+        public List<Animal> Animals { get; private set; }
```

Comment:
There is no need to have these public, and most likely should not be. Even with private set's - both can still be modified due to how memory point(s) work. Though changing the int would take some work, at least for "Animals" it is very very easy to do, I could do for example `animalShelter.Animals.Clear();`.

Make sure to only expose state that you *expect* to be used outside of the class. If it is for internal only, you may only need variables. Where you want to use properties are for creating rules around using the state or if the state needs duplicated logic. 

In your design as it stands, I think they would be fine as private variables.